src/app/inventory/store/sockets.ts

```diff
@@ -91,7 +92,14 @@ export function buildSockets(
   // get sockets from the item definition.
   if (!sockets && itemDef.sockets) {
     // If this really *should* have live sockets, but didn't...
-    if (item.itemInstanceId && item.itemInstanceId !== '0' && !socketData) {
+    if (
+      item.itemInstanceId &&
+      // a nice long instanceId is a real one and "should" have this data
+      // (short is actually a vendor item index. we'll let that build from defs.)
+      item.itemInstanceId.length > 14 &&
```

Comment:
The heuristic makes me uncomfortable, but I'm not sure of a better one.

Reply:
the conversion for vendor items, from just a number(as a string) to `${vendorHash}-${saleIndex}` happens later in item build unfortunately, or it would be more identifiable

item ids start at 69 sextillion, 19 digits, even in d1, and count up, so this is not likely to fail any time soon